### PR TITLE
Updated cloudwatch Dashboard for full set

### PR DESCRIPTION
**Pull request by veldotshine**
Updated author stack and chaos-monkey to export output variable
Cloudwatch dashboard to display CPU, Memory, Disk, SQS, ELB metrics

### CODE REVIEW

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml [hunk 1]
```diff
@@ -15,37 +15,429 @@ Resources:
...
         }
       DashboardName: !Sub
         - '${MainStackPrefixParameter}-monitoring-dashboard'
         - {
             MainStackPrefixParameter: !Ref MainStackPrefixParameter
           }
+
+
+Outputs:
+  applicationCloudwatchDashboardURL:
```
Comment (cliffano) [inline templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Other stacks' outputs keys seem to be using title case. While `applicationCloudWatchDashboardURL` is camel case. Can this please be made consistent?

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml [hunk 2]
```diff
@@ -15,37 +15,429 @@ Resources:
...
+              "y": 39,
+              "width": 12,
+              "height": 6,
+              "properties": {
+                "metrics": [
+                  [ "collectd", "GenericJMX.delay.seconds_since_last_success", "FixedDimension", "${stackPrefix}-author-standby", "Host", "${authorStandbyInstance}", "PluginInstance", "standby-status", { "period": 60, "stat": "Maximum" } ]
+                ],
+                "view": "timeSeries",
+                "stacked": false,
+                "region": "ap-southeast-2",
```
Comment (cliffano) [inline templates/cloudformation/apps/aem/full-set/monitoring.yaml]: This region should be coming from parameter just like the example.

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml [hunk 3]
```diff
@@ -15,37 +15,429 @@ Resources:
...
+            {
+              "type": "metric",
+              "x": 0,
+              "y": 39,
+              "width": 12,
+              "height": 6,
+              "properties": {
+                "view": "timeSeries",
+                "stacked": false,
+                "region": "ap-southeast-2",
```
Comment (cliffano) [inline templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Region should come from parameter/variable.

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml [hunk 4]
```diff
@@ -15,37 +15,429 @@ Resources:
...
+                  [ "...", "${authorStandbyInstance}", "Filesystem", "/dev/xvda2" ],
+                  [ "System/Linux", "DiskSpaceUtilization", "MountPath", "/", "AutoScalingGroupName", "${authorDispatcherAutoScalingGroupName}", "Filesystem", "/dev/xvda2" ],
+                  [ "...", "${publishAutoScalingGroupName}", "Filesystem", "/dev/xvda2" ],
+                  [ "...", "${publishDispatcherAutoScalingGroupName}", "Filesystem", "/dev/xvda2" ],
+                  [ "...", "${orchestratorAutoScalingGroupName}", "Filesystem", "/dev/xvda2" ],
+                  [ "...", "${choasMonkeyAutoScalingGroupName}", "Filesystem", "/dev/xvda2" ]
+                ],
+                "view": "timeSeries",
+                "stacked": false,
+                "region": "ap-southeast-2",
```
Comment (cliffano) [inline templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Region should come from parameter/variable.

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml [hunk 5]
```diff
@@ -15,37 +15,429 @@ Resources:
...
+                  [ "...", "${authorStandbyInstance}", "Filesystem", "/dev/xvdb" ],
+                  [ "System/Linux", "DiskSpaceUtilization", "MountPath", "/mnt/ebs1", "AutoScalingGroupName", "${authorDispatcherAutoScalingGroupName}", "Filesystem", "/dev/xvdb" ],
+                  [ "...", "${publishAutoScalingGroupName}", "Filesystem", "/dev/xvdb" ],
+                  [ "...", "${publishDispatcherAutoScalingGroupName}", "Filesystem", "/dev/xvdb" ],
+                  [ "...", "${orchestratorAutoScalingGroupName}", "Filesystem", "/dev/xvdb" ],
+                  [ "...", "${choasMonkeyAutoScalingGroupName}", "Filesystem", "/dev/xvdb" ]
+                ],
+                "view": "timeSeries",
+                "stacked": false,
+                "region": "ap-southeast-2",
```
Comment (cliffano) [inline templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Region should come from parameter/variable.

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml [hunk 6]
```diff
@@ -15,37 +15,429 @@ Resources:
...
+              "height": 3,
+              "properties": {
+                "metrics": [
+                  [ "AWS/ELB", "RequestCount", "LoadBalancerName", "${authorELB}" ],
+                  [ "AWS/ELB", "RequestCount", "LoadBalancerName", "${authorDispatcherELB}" ],
+                  [ "AWS/ELB", "RequestCount", "LoadBalancerName", "${publishDispatcherELB}" ]
+                ],
+                "view": "timeSeries",
+                "stacked": false,
+                "region": "ap-southeast-2",
```
Comment (cliffano) [inline templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Region should come from parameter/variable.

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml [hunk 7]
```diff
@@ -15,37 +15,429 @@ Resources:
...
+              "height": 3,
+              "properties": {
+                "metrics": [
+                  [ "AWS/ELB", "RequestCount", "LoadBalancerName", "${authorELB}" ],
+                  [ "AWS/ELB", "RequestCount", "LoadBalancerName", "${authorDispatcherELB}" ],
+                  [ "AWS/ELB", "RequestCount", "LoadBalancerName", "${publishDispatcherELB}" ]
+                ],
+                "view": "singleValue",
+                "stacked": false,
+                "region": "ap-southeast-2",
```
Comment (cliffano) [inline templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Region should come from parameter/variable.

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml [hunk 8]
```diff
@@ -15,37 +15,429 @@ Resources:
...
+              "height": 3,
+              "properties": {
+                "metrics": [
+                  [ "AWS/ELB", "HTTPCode_Backend_5XX", "LoadBalancerName", "${authorELB}" ],
+                  [ "AWS/ELB", "HTTPCode_Backend_5XX", "LoadBalancerName", "${authorDispatcherELB}" ],
+                  [ "AWS/ELB", "HTTPCode_Backend_5XX", "LoadBalancerName", "${publishDispatcherELB}" ]
+                ],
+                "view": "singleValue",
+                "stacked": false,
+                "region": "ap-southeast-2",
```
Comment (cliffano) [inline templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Region should come from parameter/variable.

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml [hunk 9]
```diff
@@ -15,37 +15,429 @@ Resources:
...
+              "height": 3,
+              "properties": {
+                "metrics": [
+                  [ "AWS/ELB", "HTTPCode_Backend_4XX", "LoadBalancerName", "${authorELB}" ],
+                  [ "AWS/ELB", "HTTPCode_Backend_4XX", "LoadBalancerName", "${authorDispatcherELB}" ],
+                  [ "AWS/ELB", "HTTPCode_Backend_4XX", "LoadBalancerName", "${publishDispatcherELB}" ]
+                ],
+                "view": "timeSeries",
+                "stacked": false,
+                "region": "ap-southeast-2",
```
Comment (cliffano) [inline templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Region should come from parameter/variable.

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml [hunk 10]
```diff
@@ -15,37 +15,429 @@ Resources:
...
+              "height": 3,
+              "properties": {
+                "metrics": [
+                  [ "AWS/ELB", "HTTPCode_Backend_4XX", "LoadBalancerName", "${authorELB}" ],
+                  [ "AWS/ELB", "HTTPCode_Backend_4XX", "LoadBalancerName", "${authorDispatcherELB}" ],
+                  [ "AWS/ELB", "HTTPCode_Backend_4XX", "LoadBalancerName", "${publishDispatcherELB}" ]
+                ],
+                "view": "singleValue",
+                "stacked": false,
+                "region": "ap-southeast-2",
```
Comment (cliffano) [inline templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Region should come from parameter/variable.

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml [hunk 11]
```diff
@@ -15,37 +15,429 @@ Resources:
...
+              "height": 3,
+              "properties": {
+                "metrics": [
+                  [ "AWS/ELB", "BackendConnectionErrors", "LoadBalancerName", "${authorELB}" ],
+                  [ "AWS/ELB", "BackendConnectionErrors", "LoadBalancerName", "${authorDispatcherELB}" ],
+                  [ "AWS/ELB", "BackendConnectionErrors", "LoadBalancerName", "${publishDispatcherELB}" ]
+                ],
+                "view": "timeSeries",
+                "stacked": false,
+                "region": "ap-southeast-2",
```
Comment (cliffano) [inline templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Region should come from parameter/variable.

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml [hunk 12]
```diff
@@ -15,37 +15,429 @@ Resources:
...
+              "height": 3,
+              "properties": {
+                "metrics": [
+                  [ "AWS/ELB", "BackendConnectionErrors", "LoadBalancerName", "${authorELB}" ],
+                  [ "AWS/ELB", "BackendConnectionErrors", "LoadBalancerName", "${authorDispatcherELB}" ],
+                  [ "AWS/ELB", "BackendConnectionErrors", "LoadBalancerName", "${publishDispatcherELB}" ]
+                ],
+                "view": "singleValue",
+                "stacked": false,
+                "region": "ap-southeast-2",
```
Comment (cliffano) [inline templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Region should come from parameter/variable.

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml [hunk 13]
```diff
@@ -15,37 +15,429 @@ Resources:
...
+              "height": 3,
+              "properties": {
+                "metrics": [
+                  [ "AWS/ELB", "UnHealthyHostCount", "LoadBalancerName", "${authorELB}" ],
+                  [ "AWS/ELB", "UnHealthyHostCount", "LoadBalancerName", "${authorDispatcherELB}" ],
+                  [ "AWS/ELB", "UnHealthyHostCount", "LoadBalancerName", "${publishDispatcherELB}" ]
+                ],
+                "view": "timeSeries",
+                "stacked": false,
+                "region": "ap-southeast-2",
```
Comment (cliffano) [inline templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Region should come from parameter/variable.

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml [hunk 14]
```diff
@@ -15,37 +15,429 @@ Resources:
...
+              "height": 3,
+              "properties": {
+                "metrics": [
+                  [ "AWS/ELB", "UnHealthyHostCount", "LoadBalancerName", "${authorELB}" ],
+                  [ "AWS/ELB", "UnHealthyHostCount", "LoadBalancerName", "${authorDispatcherELB}" ],
+                  [ "AWS/ELB", "UnHealthyHostCount", "LoadBalancerName", "${publishDispatcherELB}" ]
+                ],
+                "view": "singleValue",
+                "stacked": false,
+                "region": "ap-southeast-2",
```
Comment (cliffano) [inline templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Region should come from parameter/variable.

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml [hunk 15]
```diff
@@ -15,37 +15,429 @@ Resources:
...
+              "height": 3,
+              "properties": {
+                "metrics": [
+                  [ "AWS/ELB", "HealthyHostCount", "LoadBalancerName", "${authorELB}" ],
+                  [ "AWS/ELB", "HealthyHostCount", "LoadBalancerName", "${authorDispatcherELB}" ],
+                  [ "AWS/ELB", "HealthyHostCount", "LoadBalancerName", "${publishDispatcherELB}" ]
+                ],
+                "view": "timeSeries",
+                "stacked": false,
+                "region": "ap-southeast-2",
```
Comment (cliffano) [inline templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Region should come from parameter/variable.

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml [hunk 16]
```diff
@@ -15,37 +15,429 @@ Resources:
...
+              "height": 3,
+              "properties": {
+                "metrics": [
+                  [ "AWS/ELB", "HealthyHostCount", "LoadBalancerName", "${authorELB}" ],
+                  [ "AWS/ELB", "HealthyHostCount", "LoadBalancerName", "${authorDispatcherELB}" ],
+                  [ "AWS/ELB", "HealthyHostCount", "LoadBalancerName", "${publishDispatcherELB}" ]
+                ],
+                "view": "singleValue",
+                "stacked": false,
+                "region": "ap-southeast-2",
```
Comment (cliffano) [inline templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Region should come from parameter/variable.

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml [hunk 17]
```diff
@@ -15,37 +15,429 @@ Resources:
...
+                  [ "...", "${authorStandbyInstance}", { "period": 300, "label": "Author Dispatcher", "stat": "Maximum" } ],
+                  [ "System/Linux", "MemoryUtilization", "AutoScalingGroupName", "${authorDispatcherAutoScalingGroupName}", { "period": 300, "label": "Author Dispatcher", "stat": "Maximum" } ],
+                  [ "...", "${publishAutoScalingGroupName}", { "period": 300, "label": "Publish", "stat": "Maximum" } ],
+                  [ "...", "${publishDispatcherAutoScalingGroupName}", { "period": 300, "label": "Publish Dispatcher", "stat": "Maximum" } ],
+                  [ "...", "${choasMonkeyAutoScalingGroupName}", { "period": 300, "label": "Choas Monkey", "stat": "Maximum" } ],
+                  [ "...", "${orchestratorAutoScalingGroupName}", { "period": 300, "label": "Orchestrator", "stat": "Maximum" } ]
+                ],
+                "view": "singleValue",
+                "stacked": false,
+                "region": "ap-southeast-2",
```
Comment (cliffano) [inline templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Region should come from parameter/variable.

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml [hunk 18]
```diff
@@ -15,37 +15,429 @@ Resources:
...
+                  [ "...", "${authorStandbyInstance}", { "period": 300, "label": "Author Dispatcher", "stat": "Maximum" } ],
+                  [ "System/Linux", "MemoryUtilization", "AutoScalingGroupName", "${authorDispatcherAutoScalingGroupName}", { "period": 300, "label": "Author Dispatcher", "stat": "Maximum" } ],
+                  [ "...", "${publishAutoScalingGroupName}", { "period": 300, "label": "Publish", "stat": "Maximum" } ],
+                  [ "...", "${publishDispatcherAutoScalingGroupName}", { "period": 300, "label": "Publish Dispatcher", "stat": "Maximum" } ],
+                  [ "...", "${choasMonkeyAutoScalingGroupName}", { "period": 300, "label": "Choas Monkey", "stat": "Maximum" } ],
+                  [ "...", "${orchestratorAutoScalingGroupName}", { "period": 300, "label": "Orchestrator", "stat": "Maximum" } ]
+                ],
+                "view": "timeSeries",
+                "stacked": false,
+                "region": "ap-southeast-2",
```
Comment (cliffano) [inline templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Region should come from parameter/variable.

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml [hunk 19]
```diff
@@ -15,37 +15,429 @@ Resources:
...
+            },
+            {
+              "type": "metric",
+              "x": 0,
+              "y": 33,
+              "width": 12,
+              "height": 6,
+              "properties": {
+                "metrics": [
+                  [ "System/Linux", "DiskSpaceUtilization", "MountPath", "/mnt/ebs1", "InstanceId", "${authorPrimaryInstance}", "Filesystem", "/dev/xvdb" ],
```
Comment (cliffano) [inline templates/cloudformation/apps/aem/full-set/monitoring.yaml]: All `/dex/xvdb` should come from parameter data volume.

Comment (cliffano) [inline templates/cloudformation/apps/aem/full-set/monitoring.yaml]: All `/mnt/ebs1` should come from mount point config.

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml [hunk 20]
```diff
@@ -15,37 +15,429 @@ Resources:
...
+            {
+              "type": "metric",
+              "x": 12,
+              "y": 33,
+              "width": 12,
+              "height": 6,
+              "properties": {
+                "metrics": [
+                  [ "System/Linux", "DiskSpaceUtilization", "MountPath", "/", "InstanceId", "${authorPrimaryInstance}", "Filesystem", "/dev/xvda2" ],
+                  [ "...", "${authorStandbyInstance}", "Filesystem", "/dev/xvda2" ],
```
Comment (cliffano) [inline templates/cloudformation/apps/aem/full-set/monitoring.yaml]: All `/dev/xvda2` should be parameterised.

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml (hunk 1 above)
Comment (veldotshine) [reply templates/cloudformation/apps/aem/full-set/monitoring.yaml]: replaced with `ApplicationCloudWatchDashboardURL`.

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml (hunk 2 above)
Comment (veldotshine) [reply templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Thanks for highlighting. Replaced region value with variable `${awsRegion}`

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml (hunk 4 above)
Comment (veldotshine) [reply templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Replaced region value with variable ${awsRegion}.

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml (hunk 3 above)
Comment (veldotshine) [reply templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Replaced region value with variable ${awsRegion}

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml (hunk 5 above)
Comment (veldotshine) [reply templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Replaced region value with variable ${awsRegion}

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml (hunk 6 above)
Comment (veldotshine) [reply templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Replaced region value with variable ${awsRegion}

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml (hunk 7 above)
Comment (veldotshine) [reply templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Replaced region value with variable ${awsRegion}

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml (hunk 8 above)
Comment (veldotshine) [reply templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Replaced region value with variable ${awsRegion}

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml (hunk 9 above)
Comment (veldotshine) [reply templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Replaced region value with variable ${awsRegion}

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml (hunk 10 above)
Comment (veldotshine) [reply templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Replaced region value with variable ${awsRegion}

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml (hunk 11 above)
Comment (veldotshine) [reply templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Replaced region value with variable ${awsRegion}

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml (hunk 12 above)
Comment (veldotshine) [reply templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Replaced region value with variable ${awsRegion}

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml (hunk 13 above)
Comment (veldotshine) [reply templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Replaced region value with variable ${awsRegion}

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml (hunk 14 above)
Comment (veldotshine) [reply templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Replaced region value with variable ${awsRegion}

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml (hunk 15 above)
Comment (veldotshine) [reply templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Replaced region value with variable ${awsRegion}

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml (hunk 16 above)
Comment (veldotshine) [reply templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Replaced region value with variable ${awsRegion}

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml (hunk 17 above)
Comment (veldotshine) [reply templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Replaced region value with variable ${awsRegion}

File: templates/cloudformation/apps/aem/full-set/monitoring.yaml (hunk 18 above)
Comment (veldotshine) [reply templates/cloudformation/apps/aem/full-set/monitoring.yaml]: Replaced region value with variable ${awsRegion}

**Pull request merged**